tests/test_func.py

```diff
@@ -77,8 +72,7 @@ def _test(self, tocheck: List[str]) -> None:
         try:
             self.linter.check(tocheck)
         except Exception as ex:
-            msg = f"Exception: {ex} ({exception_str} while checking {tocheck}))"
-            raise Exception(msg) from ex
+            raise Exception from ex
```

Comment:
@Pierre-Sassoulas Is this what you meant?

Reply:
I'm thinking of:
```diff
diff --git a/tests/test_func.py b/tests/test_func.py
index b57dc7d1..ba13844f 100644
--- a/tests/test_func.py
+++ b/tests/test_func.py
@@ -36,10 +36,6 @@ FILTER_RGX = None
 INFO_TEST_RGX = re.compile(r"^func_i\d\d\d\d$")
 
 
-def exception_str(self, ex):  # pylint: disable=unused-argument
-    """function used to replace default __str__ method of exception instances"""
-    return f"in {ex.file}\n:: {', '.join(ex.args)}"
-
 
 class LintTestUsingModule:
     INPUT_DIR = None
@@ -75,12 +71,11 @@ class LintTestUsingModule:
         try:
             self.linter.check(tocheck)
         except Exception as ex:
+            msg = f"Exception: {ex} in {tocheck}\n:: {', '.join(ex.args)})"
+            raise Exception(msg) from ex
+        finally:
             # need finalization to restore a correct state
             self.linter.reporter.finalize()
-            ex.file = tocheck
-            print(ex)
-            ex.__str__ = exception_str
-            raise
         self._check_result(self.linter.reporter.finalize())
 
     def _has_output(self):
```

Reply:
I assumed the ) add the end of the f-string was unintentional, please correct if I'm wrong. Going through tests now.

Reply:
Yes you're probably right. This is just a piece of code helpful for pylint's contributor during development it does not need to be perfect :)

Reply:
Does fail a lot of test though 😢

Reply:
Is this change making the test fail ? We can just ignore the type error and not change anything if that's the case.

Reply:
Yeah, see test history of latest three commits. However, I think it is due to some formatting changes which can be auto solved by pytest. I'm on mobile now so I can't actually run pytest.

Reply:
Ok I'll try something then.

Reply:
Hurgh, just realized that those are the legacy functional tests, we're trying to get rid of this code. Let's not do anything at all, ignore all the error and not add typing if it's causing trouble.  I think it would probably be faster to finish the migration to the new functional tests framework than fix this. See :
```python
    assert len(tests) < 13, "Please do not add new test cases here." + "\n".join(
        str(k) for k in tests if not k[2]
    )
```
There's not much left of them too.

Reply:
Done!